kwave/options/simulation_execution_options.py

```diff
@@ -179,6 +188,27 @@ def as_list(self, sensor: kSensor) -> list[str]:
             options_list.append("--verbose")
             options_list.append(str(self.verbose_level))
 
+        if (self.checkpoint_interval is not None or self.checkpoint_timesteps is not None) and self.checkpoint_file is not None:
```

Comment:
Perhaps you could set a default checkpointing path, or warn the user if checkpoint_interval or checkpoint_timesteps are set, but not checkpoint_file? As it is now, not passing a checkpoint file would simulate without checkpointing and not warn the user, which might be confusing.

c.f. Updating the example with these execution options.
```python
  execution_options = SimulationExecutionOptions(checkpoint_timesteps=checkpoint_timesteps)
  ```

Reply:
I understand your sentiment here. Perhaps the other checkpointing parameters should validate that the user has set the path and throw an error if not.